### PR TITLE
add functions to remove notifications by prefix/suffix

### PR DESCRIPTION
**Pull request by dvaldivia**
Introduces functions to be able to remove bucket notifications by arn+events+prefix+suffix

- RemoveTopicByArnEventsPrefixSuffix
- RemoveQueueByArnEventsPrefixSuffix
- RemoveLambdaByArnEventsPrefixSuffix

### CODE REVIEW

File: bucket-notification.go [hunk 1]
```diff
@@ -137,6 +137,74 @@ func (t *NotificationConfig) AddFilterPrefix(prefix string) {
...
+	for _, i := range a {
+		setA.Add(string(i))
+	}
+
+	setB := set.NewStringSet()
+	for _, i := range b {
+		setB.Add(string(i))
+	}
+
+	if len(setA.Intersection(setB)) == len(a) {
```
Comment (harshavardhana) [inline bucket-notification.go:155]: You can help .IsEmpty()

File: bucket-notification.go [hunk 2]
```diff
@@ -137,6 +137,74 @@ func (t *NotificationConfig) AddFilterPrefix(prefix string) {
...
+	for _, i := range a {
+		setA.Add(fmt.Sprintf("%s-%s", i.Name, i.Value))
+	}
+
+	setB := set.NewStringSet()
+	for _, i := range b {
+		setB.Add(fmt.Sprintf("%s-%s", i.Name, i.Value))
+	}
+
+	if len(setA.Intersection(setB)) == len(a) {
```
Comment (harshavardhana) [inline bucket-notification.go:177]: You can use IsEmpty()

File: bucket-notification.go (hunk 1 above)
Comment (dvaldivia) [reply bucket-notification.go:155]: but if both sets have the same elements, wouldn't the intersection be a new set of the same size?

File: bucket-notification.go (hunk 2 above)
Comment (dvaldivia) [reply bucket-notification.go:177]: I'll do `setA.Difference(setB).IsEmpty()` then

File: bucket-notification.go [hunk 3]
```diff
@@ -137,6 +137,73 @@ func (t *NotificationConfig) AddFilterPrefix(prefix string) {
...
+	for _, i := range a {
+		setA.Add(string(i))
+	}
+
+	setB := set.NewStringSet()
+	for _, i := range b {
+		setB.Add(string(i))
+	}
+
+	if setA.Difference(setB).IsEmpty() {
```
Comment (harshavardhana) [inline bucket-notification.go]: The if condition can be combined now as one

```
return setA.Difference(setB).IsEmpty()
```

File: bucket-notification.go [hunk 4]
```diff
@@ -137,6 +137,73 @@ func (t *NotificationConfig) AddFilterPrefix(prefix string) {
...
+	for _, i := range a {
+		setA.Add(fmt.Sprintf("%s-%s", i.Name, i.Value))
+	}
+
+	setB := set.NewStringSet()
+	for _, i := range b {
+		setB.Add(fmt.Sprintf("%s-%s", i.Name, i.Value))
+	}
+
+	if setA.Difference(setB).IsEmpty() {
```
Comment (harshavardhana) [inline bucket-notification.go]: Single line

```
return setA.Difference(setB).IsEmpty()
```

File: bucket-notification_test.go [hunk 5]
```diff
@@ -0,0 +1,216 @@
+// This file is part of MinIO Console Server
```
Comment (harshavardhana) [inline bucket-notification_test.go:1]: Wrong copyright @dvaldivia

**Comment by dvaldivia** [comment]
@harshavardhana fixed the copyright

File: bucket-notification.go [hunk 6]
```diff
@@ -19,7 +19,7 @@ package minio
 
 import (
 	"encoding/xml"
-
+	"fmt"
```
Comment (krisis) [inline bucket-notification.go:22]: License header requires an update

File: bucket-notification_test.go [hunk 7]
```diff
@@ -0,0 +1,217 @@
+/*
+ * MinIO Go Library for Amazon S3 Compatible Cloud Storage
+ * Copyright 2015-2017 MinIO, Inc.
```
Comment (krisis) [inline bucket-notification_test.go:3]:
```suggestion
 * Copyright 2015-2020 MinIO, Inc.
```

File: bucket-notification.go (hunk 6 above)
Comment (dvaldivia) [reply bucket-notification.go:22]: done!

**Pull request merged**